gobblin-runtime/src/main/java/gobblin/runtime/JobState.java

```diff
@@ -122,6 +123,7 @@ public boolean isRunningOrDone() {
   private RunningState state = RunningState.PENDING;
   private int taskCount = 0;
   private final Map<String, TaskState> taskStates = Maps.newLinkedHashMap();
+  private final Map<String, TaskState> skippedTaskStates = Maps.newLinkedHashMap();
```

Comment:
Could you put in a brief comment here about why this was needed. As in this is required to avoid reaching the publisher, but still allow state persistence.